src/py42/services/_connection.py

```diff
@@ -76,6 +76,19 @@ def get_host_address(self):
         return self._kv_service.get_stored_value(self._key).text
 
 
+class ConnectedServerHostResolver(HostResolver):
+    def __init__(self, connection, device_guid):
+        self._connection = connection
+        self._device_guid = device_guid
+        super().__init__()
+
+    def get_host_address(self):
+        response = self._connection.get(
+            u"api/connectedServerUrl", params={u"guid": self._device_guid}
```

Comment:
I don't recall needing anything like this the first time I attempted doing push restore via the API. Has something changed? I do remember that:
- there were some endpoints that I thought should be called on the storage node that were actually called on the authority instead
- it was important that some of the api calls contain `/console/` in the url in order to force the load balancer to have us stick with a given server.

Reply:
The only way I was able
to get the API's to work
was this way or to use some cookie in the browser with c42_auth, I think something did change.

Reply:
This is a push-restore-only thing. Since the client you are pushing to has to be online to receive the restore request, we have to find out if it's actively online, which is what `connectedServerUrl` does. The UI does this behind the scenes when you put in the target guid and doesn't let you start the restore if the client is offline.

Reply:
> The only way I was able
> to get the API's to work
> was this way or to use some cookie in the browser with c42_auth, I think something did change.

This read like a poem, beautiful stuff

Reply:
> This is a push-restore-only thing. Since the client you are pushing to has to be online to receive the restore request, we have to find out if it's actively online, which is what `connectedServerUrl` does. The UI does this behind the scenes when you put in the target guid and doesn't let you start the restore if the client is offline.

Cool, makes sense to me